src/library/trackcollection.cpp

```diff
@@ -86,7 +85,7 @@ void TrackCollection::connectDatabase(QSqlDatabase database) {
 }
 
 void TrackCollection::disconnectDatabase() {
-    DEBUG_ASSERT(QApplication::instance()->thread() == QThread::currentThread());
+    DEBUG_ASSERT_QOBJECT_THREAD_AFFINITY(this);
```

Comment:
Did you concider to add a version that omitt "this"?
This would make the null check unnecessary and the "this" dereference inside the macro.

Reply:
Explicitly passing `this` makes it more obvious on which instance this macro operates on.

Reply:
Yes, but this comes with some the cost of an extra null check. Isn't it obvious enough without this? What else could it check?

Reply:
The extra null check in a debug assertion is irrelevant.

I dislike functions and macros with implicit parameters that solely depend on their call context, because they are more difficult to reason about.

Reply:
OK